pkgs/development/python-modules/notify-py/default.nix

```diff
@@ -12,22 +13,27 @@ buildPythonPackage rec {
   };
 
   postPatch = ''
-   substituteInPlace setup.py \
-     --replace "loguru==0.4.1" "loguru~=0.5.0"
+    substituteInPlace setup.py \
+      --replace "loguru==0.4.1" "loguru~=0.5.0"
   '';
 
-  propagatedBuildInputs = [ alsaUtils libnotify loguru which ];
+  propagatedBuildInputs = [ loguru which ]
+    ++ lib.optionals stdenv.isLinux [ alsaUtils libnotify ];
 
-  checkInputs = [ alsaUtils libnotify pytest which ];
+  checkInputs = [ pytest which ]
+    ++ lib.optionals stdenv.isLinux [ alsaUtils libnotify ];
```

Comment:
```suggestion
  checkInputs = [ pytestCheckHook ];
```

Reply:
If I do this the tests actually fail.

```
E           notifypy.exceptions.BinaryNotFound: Unable to find required notify-send. Please check if it's installed.
```

Reply:
Then the final package should also not work correct and miss notify-send. Maybe we need to patch the path to the binary?

Reply:
Seems to work fine:

```sh
❯ result/bin/notifypy --title test --message test --applicationName test --soundPath ~/Downloads/glass_ping-Go445-1207030150.wav
```

On Gnome I see the popup notification an hear the sound.

Reply:
I do have `aplay` available on `PATH`, maybe it's succeeding because of that?

Reply:
As a test removed `alsaUtils` and `libnotify` from `propagatedBuildInputs` and it then fails to run.

```sh
❯ result/bin/notifypy --title test --message test --applicationName test --soundPath ~/Downloads/glass_ping-Go445-1207030150.wav
which: no notify-send in (/nix/store/d44wd6n98f93hjr6q1d1phhh1hw7a17d-python3-3.8.8/bin:/nix/store/dhp5sw5g7b35xxv5d2411m0h1qckl495-python3.8-notify-py-0.3.1/bin:/nix/store/c0i69hngm020s610kkh6jlhgjjvn6w12-which-2.21/bin:/run/wrappers/bin:/home/austin/.local/share/flatpak/exports/bin:/var/lib/flatpak/exports/bin:/home/austin/.nix-profile/bin:/etc/profiles/per-user/austin/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin)
Traceback (most recent call last):
  File "/nix/store/dhp5sw5g7b35xxv5d2411m0h1qckl495-python3.8-notify-py-0.3.1/bin/.notifypy-wrapped", line 9, in <module>
    sys.exit(entry())
  File "/nix/store/dhp5sw5g7b35xxv5d2411m0h1qckl495-python3.8-notify-py-0.3.1/lib/python3.8/site-packages/notifypy/cli.py", line 86, in entry
    Notify(
  File "/nix/store/dhp5sw5g7b35xxv5d2411m0h1qckl495-python3.8-notify-py-0.3.1/lib/python3.8/site-packages/notifypy/notify.py", line 73, in __init__
    self._notifier = self._notifier_detect(**kwargs)
  File "/nix/store/dhp5sw5g7b35xxv5d2411m0h1qckl495-python3.8-notify-py-0.3.1/lib/python3.8/site-packages/notifypy/os_notifiers/linux.py", line 22, in __init__
    raise BinaryNotFound("notify-send")
notifypy.exceptions.BinaryNotFound: Unable to find required notify-send. Please check if it's installed.
```

Reply:
Yeah, I think this is still wrong, but this does at least fix the build for Darwin.

Reply:
@lukegb any idea or suggestions?

Reply:
My suggestion would be dropping these non-Python deps from propagatedBuildInputs and checkInputs, and then applying a Linux-only patch to swap out https://github.com/ms7m/notify-py/blob/master/notifypy/os_notifiers/linux.py#L33-L59 for functions which just statically return the paths to the binaries (i.e. "${alsaUtils}/bin/aplay" and "${libnotify}/bin/notify-send")

Reply:
For an example of this, see https://sourcegraph.com/github.com/NixOS/nixpkgs/-/blob/pkgs/development/python-modules/sphinxcontrib-tikz/default.nix#L19-24

Reply:
I think I've sorted this out now without requiring patching.